### PR TITLE
Add support for ordering projects by deployment status

### LINKED ISSUE #4312: Make it possible to sort projects by status
Currently "Status" column is an outcome of `asset.has_deployment` and `asset.deployment__active`. Using these two fields combined we can say if project is Draft, Deployed, or Archived. Given these are two separate fields it is not possible to sort by "Status" column. We would need a way to sort on both `/api/v2/project-views/${viewUid}/assets/` and `/api/v2/assets/` endpoints.

### PR DESCRIPTION
**Pull request by noliveleger**
## Checklist

1. [x] If you've added code that should be tested, add tests
2. [x] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a description of your work suitable for publishing on [our forum](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Make the UI (and the API) sortable by deployment status

## Additional info 

The attribute returned by the API is `deployment_status` whereas the model field is `_deployment_status`. Therefore, to use the query parser or to order by deployment status, use `_deployment_status` (with leading underscore). 
E.g: 
- Sort: `/api/v2/assets.json?ordering=_deployment_status`
- Search: `/api/v2/assets.json?q=_deployment_status:deployed`

## Related issues

Closes #4312

### CODE REVIEW

**Review by magicznyleszek — APPROVED** [review]

**Comment by magicznyleszek** [comment]
@noliveleger I've pushed few small tweaks and approved this PR :)

**Comment by noliveleger** [comment]
> @noliveleger I've pushed few small tweaks and approved this PR :)

I expected no less from you 😘 . Thank you.

**Comment by JacquelineMorrissette** [comment]
@magicznyleszek Should Description be included here? This is one of the fields that has to be configured in Constance to become visible anywhere else
image

**Comment by JacquelineMorrissette** [comment]
Just a small UI bug, I suspect these should be on the same line
image

**Review by JacquelineMorrissette — APPROVED** [review]

File: kpi/models/asset.py [hunk 1]
```diff
@@ -82,6 +82,13 @@
 from kpi.tasks import remove_asset_snapshots
 
 
+class AssetDeploymentStatus(models.TextChoices):
+
+    ARCHIVED = 'archived', 'Archived'
+    DEPLOYED = 'deployed', 'Deployed'
+    DRAFT = 'draft', 'Draft'
```
[the remark this answers is not in the document]
Comment (noliveleger) [reply kpi/models/asset.py:89]: Front end uses its own labels. AFAIK These ones don't need to be translated.

**Pull request merged**